src/ds/d500/d500-device.cpp

```diff
@@ -618,6 +618,20 @@ namespace librealsense
                 std::make_shared<temperature_option>(_hw_monitor,
                     temperature_option::temperature_component::LEFT_PROJ, "Temperature reading for Left Projector"));
 
+            auto error_control = std::make_shared< uvc_xu_option< uint8_t > >( raw_depth_sensor,
+                                                                               depth_xu,
+                                                                               DS5_ERROR_REPORTING,
+                                                                               "Error reporting" );
+
+            _polling_error_handler
+                = std::make_shared< polling_error_handler >( 1000,
+                                                             error_control,
+                                                             raw_depth_sensor->get_notifications_processor(),
+                                                             std::make_shared< d500_notification_decoder >() );
+
+            depth_sensor.register_option( RS2_OPTION_ERROR_POLLING_ENABLED,
```

Comment:
@maloel is it OK to register for all D500 currently?
Or maybe 5E didn't implement it yet and you will have issues with it?

Reply:
I don't know if it's implemented.
It's likely we'll need it for USB. But USB is not our priority. We can try... add it and I'll see if it works... @_@

Reply:
I figured that since different D500  models will use same HKR FW it will work, but you are right that 5E might need some handling to pass it to the user.